workflows/prognostic_c48_run/runtime/factories.py

```diff
@@ -24,3 +27,37 @@ def get_emulator_adapter(
             diagnostic_variables=set(config.diagnostic_variables),
             timestep=timestep,
         )
+
+
+def get_overrider_adapter(
+    config: UserConfig,
+    state: DerivedFV3State,
+    communicator: fv3gfs.util.CubedSphereCommunicator,
+    timestep: float,
+) -> Optional[OverriderAdapter]:
+    if config.overrider is None:
+        return None
+    else:
+        return OverriderAdapter(
+            config.overrider,
+            state,
+            communicator,
+            timestep,
+            diagnostic_variables=set(config.diagnostic_variables),
+        )
+
+
+def get_fv3_physics_adapter(
+    config: UserConfig,
+    state: DerivedFV3State,
+    timestep: float,
+    communicator: fv3gfs.util.CubedSphereCommunicator,
+) -> Optional[Union[PrognosticAdapter, OverriderAdapter]]:
```

Comment:
This is a very broad output type and could get confusing for readers to work out the interfaces the returned object is meant to have. Would be clearer to explicitly define the expected interface using either `typing_extensions.Protocol` or a an abstract base-class for `Adapter`. It's also possible that this leaky abstraction will result in disambiguating conditional logic in the `loop.py`, but I haven't read that far yet.

 
```suggestion
) -> Optional[Adapter]:
```

Reply:
I deleted this function.